src/lm.jl

```diff
@@ -195,44 +229,49 @@ For linear models, the deviance of the null model is equal to the total sum of s
 """
 function nulldeviance(obj::LinearModel)
     y = obj.rr.y
-    wts = obj.rr.wts
-
+    wts = obj.pp.wts
     if hasintercept(obj)
-        if isempty(wts)
-            m = mean(y)
-        else
-            m = mean(y, weights(wts))
-        end
+        m = mean(y, wts)
     else
         @warn("Starting from GLM.jl 1.8, null model is defined as having no predictor at all " *
               "when a model without an intercept is passed.")
         m = zero(eltype(y))
     end
 
     v = zero(eltype(y))*zero(eltype(wts))
-    if isempty(wts)
-        @inbounds @simd for yi in y
-            v += abs2(yi - m)
+    if wts isa UnitWeights
+        @inbounds @simd for i = eachindex(y,wts)
+            v += abs2(y[i] - m)
         end
     else
         @inbounds @simd for i = eachindex(y,wts)
             v += abs2(y[i] - m)*wts[i]
         end
     end
-    v
+    return v
+end
+
+function nullloglikelihood(m::LinearModel)
+    wts = weights(m)
+    if wts isa Union{UnitWeights, FrequencyWeights}
+        n = nobs(m)
+        -n/2 * (log(2π * nulldeviance(m)/n) + 1)
+    else
+        N = length(m.rr.y)
+        n = sum(log, wts)
+        0.5*(n - N * (log(2π * nulldeviance(m)/N) + 1))
```

Comment:
Are we sure this definition is OK for both analytical weights and probability weights? I think we discussed this before, but `loglikelihood` throws an error for probability weights so I'm surprised that `nullloglikelihood` doesn't.